mteb/models/sentence_transformer_wrapper.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from typing import Any, get_args
+
+import numpy as np
+import torch
+from sentence_transformers import CrossEncoder, SentenceTransformer
+
+import mteb
+from mteb.abstasks.TaskMetadata import TASK_TYPE
+from mteb.encoder_interface import PromptType
+
+logger = logging.getLogger(__name__)
+
+
+class SentenceTransformerWrapper:
+    def __init__(
+        self,
+        model: str | SentenceTransformer | CrossEncoder,
+        revision: str | None = None,
+        task_to_prompt_name: dict[str, str] | None = None,
+        model_prompts: dict[str, str] | None = None,
+        override_prompts: bool = False,
+        **kwargs,
+    ) -> None:
+        """Wrapper for SentenceTransformer models.
+
+        Args:
+            model: The SentenceTransformer model to use. Can be a string (model name), a SentenceTransformer model, or a CrossEncoder model.
+            revision: The revision of the model to use.
+            task_to_prompt_name: A dictionary mapping task names to prompt names.
+                First priority is given to the composed prompt of task name + prompt type (query or passage), then to the specific task prompt,
+                then to the composed prompt of task type + prompt type, then to the specific task type prompt,
+                and finally to the specific prompt type.
+            model_prompts: A dictionary mapping prompt names with prompts. If not provided, the model's prompts will be used.
+            override_prompts: Whether to override the model's existing prompts with the provided model_prompts.

```

Comment:
I would just always do this - if you specify model_prompts then I would expect that the new ones are used.